### PR TITLE
[TASK] Address recent Coding Standards

### PR DESCRIPTION
**Pull request by garvinhicking**
refs https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/4011

Releases: main, 12.4, 11.5

### CODE REVIEW

**Review by linawolf — APPROVED** [review]

**Comment by linawolf** [comment]
Should this be backported?

File: Documentation/CodingGuidelines/CglPhp/PhpFileFormatting/GeneralRequirementsForPhpFiles.rst [hunk 1]
```diff
@@ -91,7 +101,9 @@ Comment lines should be kept within a limit of about 80 characters
 Whitespace and Indentation
 ==========================
 
-TYPO3 uses space characters to indent source code. Following PSR-2,
+TYPO3 uses space characters to indent source code. Following the
+`TYPO3 Coding Standards <https://github.com/TYPO3/coding-standards>`__,
+(based on PHP-FIG, PSR-1/PSR-2 and PSR-12 at the time of this writing),
```
Comment (brotkrueml) [inline Documentation/CodingGuidelines/CglPhp/PhpFileFormatting/GeneralRequirementsForPhpFiles.rst:106]: Here too.

**Review by brotkrueml — COMMENTED** [review]
Actually, I don't know how to handle the transition phase between PER-CS1.0/PER-CS2.0 in these sections. Either use to the typo3/coding-standards package or use one of the pure standards (PER_CS1.0 does not handle new PHP stuff, like enums, while PER-CS2.0 does. But PER-CS2.0 is currently not fully implemented in php-cs-fixer, hence the transition).

**Comment by garvinhicking** [comment]
@brotkrueml I've rephrased this to try to better indicate the changing standards and be more abstract.

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**